src/include/duckdb/common/single_thread_pointer.h

```diff
@@ -0,0 +1,169 @@
+#include <cstdlib>
+
+template <typename T>
```

Comment:
This should be in the duckdb namespace and use standard header conventions (include the header comments up top)

Reply:
I see.